File: apps/reader/src/prompts/prompts.ts
import { ChatPromptTemplate } from '@langchain/core/prompts'

const SYSTEM_TEMPLATE = `Use the following pieces of context to answer the question at the end.
If you don't know the answer, just say that you don't know, don't try to make up an answer.
----------------
{context}`

export const ragPrompt = ChatPromptTemplate.fromMessages([
  ['system', SYSTEM_TEMPLATE],
  ['human', '{question}'],
])

File: apps/api/src/prompts.py
from langchain_core.prompts import ChatPromptTemplate

# RAG用のプロンプトテンプレート
rag_template = """あなたは親切で役立つAIアシスタントです。
ユーザーの質問に対して、以下のコンテキスト情報を使って回答してください。
コンテキストに含まれていない情報がある場合は、「その情報はコンテキストには含まれていません」と正直に答えてください。

コンテキスト:
{context}

質問: {question}
"""

rag_prompt = ChatPromptTemplate.from_template(rag_template)
